src/Perlang.Interpreter/Typing/TypeResolver.cs

```diff
@@ -171,7 +171,7 @@ public override VoidObject VisitBinaryExpr(Expr.Binary expr)
                              (leftTypeReference.ClrType == typeof(AsciiString) &&
                               rightTypeReference.ClrType == typeof(AsciiString))) {
                         // "string" + "string"
-                        expr.TypeReference.ClrType = typeof(Utf8String);
+                        expr.TypeReference.ClrType = typeof(AsciiString);
```

Comment:
:point_up: This was a bug; `ASCIIString+ASCIIString` should obviously not return a `UTF8String`. However, both of these would fail equally because of the missing supporting code in the C++ stdlib (fixed in this PR).